numpy/lib/financial.py

```diff
@@ -727,12 +744,17 @@ def mirr(values, finance_rate, reinvest_rate):
         Modified internal rate of return
 
     """
-    values = np.asarray(values, dtype=np.double)
+    values = np.asarray(values)
     n = values.size
+
+    # Support decimal computation
+    if isinstance(finance_rate, Decimal):
+        n = Decimal(n)
```

Comment:
Why is this needed?

Reply:
If `n` is not a decimal then the computation below `1/(n - 1)` becomes a float and causes a `TypeError`. 

`TypeError: unsupported operand type(s) for ** or pow(): 'decimal.Decimal' and 'float'`

Reply:
Hmm, you're right. I can't think of a good way around that.

On the other hand, you already say some other functions are not supported because they use `sqrt` - if those aren't, then this shouldn't be either

Reply:
Is there value in not supporting `Decimal` in places that it can be supported? That's why I've added to everywhere that I could. 

I couldn't figure out how/where `.log` and the eigen log value calculations were located. I would have added the support in those places to call the appropriate Decimal methods, however, I couldn't find it, so I couldn't add it.

Reply:
What do you mean by "eigen log value"?

I thought the problem was that python doesn't even define `log` on `Decimal` deliberately, but that's not the case. Just keep in mind the operations that python does and doesn't allow, and try to respect them. What you have here is fine

Could you expand this python comment explaining it's for the (1/n-1) below?

Reply:
I plan to make all the changes as mentioned, in this review. I appreciate the fast feedback.

I think the call to log and square root will work out fine once the ticket for using `round`, `log`, etc. for different types works. 

It seems decimal does have `.log*` methods that'd mostly do the job, but it'd have to be custom for Decimal because it has limited log support.